hack/upgrade-test-operator-sdk.sh

```diff
@@ -0,0 +1,315 @@
+#!/bin/bash -e
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2021 Red Hat, Inc.
+#
+# Usage:
+# make upgrade-test-index-image
+#
+# Use Openshift-CI "optional-operators-ci-*" workflow to:
+# - Build an internal index image based off of the index image at:
+#   quay.io/kubevirt/hyperconverged-cluster-index
+#   with the appropriate tag of the version
+# - Add to that index a new bundle, named 100.0.0 with the contents
+#   of the open PR (this can include new dependent images, new CRDs...).
+# - Subscribe to the initial channel, using the "optional-operators-ci-subscribe"
+#   step.
+# This script is then upgrading HCO to 100.0.0 version, by patching the subscripiton,
+# and performs various validations against the upgraded version.
+
+
+MAX_STEPS=$(( $(grep -c "Msg " "$0") - 2)) # subtract self line and the function name
+CUR_STEP=1
+HCO_DEPLOYMENT_NAME=hco-operator
+HCO_WH_DEPLOYMENT_NAME=hco-webhook
+HCO_NAMESPACE="kubevirt-hyperconverged"
+HCO_KIND="hyperconvergeds"
+HCO_RESOURCE_NAME="kubevirt-hyperconverged"
+TARGET_VERSION=100.0.0
+VMS_NAMESPACE=vmsns
+
+OUTPUT_DIR=${OUTPUT_DIR:-_out}
+
+echo "INITIAL_VERSION: $INITIAL_VERSION"
+
+function Msg {
+    { set +x; } 2>/dev/null
+    echo "--"
+    for a in "$@"; do
+        echo "Upgrade Step ${CUR_STEP}/${MAX_STEPS}: $a"
+    done
+    echo "--"
+    ((CUR_STEP += 1))
+    set -x
+}
+
+export CMD="oc"
+
+echo "oc version"
+${CMD} version || true
```

Comment:
why do you think `oc version` command can fail?
and if it fails because for some reason `oc` is missing, or the cluster is not reachable, i think it's a no-go for the rest of the script and shouldn't ignored silently.

Reply:
This is a copy from the source of this script - upgrade-test-index-image.sh. I guess it was added for a reason